### ISSUE TITLE
Test Explorer Picks up Zero Tests in VS 2017 for NUnit 2

### ISSUE THREAD

**Issue by DanJ210** [nunit/nunit-vs-adapter#178]
Where can I find a sample for NUnit 2? I have to use it for something but have spent the entire day trying to get it to work with zero success.

I can get NUnit 3 to work and with this sample but NUnit 2 just finds zero tests. I've installed NUnit framework 2.6.1 - 2.7 with the NUnitTestAdapter 2.0 - 2.1.1 trying all possible combinations. Below is my `[TextFixture]` and my `[TestCase]` which is an NUnit 2.0 syntax as shown below. I've searched all day and have tried every suggestion but no matter what I've not been able to get tests picked up using NUnit 2.

```
[TestFixture]
    public class UnitTest1
    {
        [TestCase]
        public void ProgrammerTest()
        {
            var address = new Address("56 Main St", "Mesa", "AZ", "38574");
            var customer = new Customer("John", "Doe", address);
            var company = new Company("Google", address);

            Assert.IsNullOrEmpty(customer.Id);
            customer.Save();
            Assert.IsNotNullOrEmpty(customer.Id);

            Assert.IsNullOrEmpty(company.Id);
            company.Save();
            Assert.IsNotNullOrEmpty(company.Id);

            Customer savedCustomer = Customer.Find(customer.Id);
            Assert.IsNotNull(savedCustomer);
            Assert.AreSame(customer.Address, address);
            Assert.AreEqual(savedCustomer.Address, address);
            Assert.AreEqual(customer.Id, savedCustomer.Id);
            Assert.AreEqual(customer.FirstName, savedCustomer.FirstName);
            Assert.AreEqual(customer.LastName, savedCustomer.LastName);
            Assert.AreEqual(customer, savedCustomer);
            Assert.AreNotSame(customer, savedCustomer);

            Company savedCompany = Company.Find(company.Id);
            Assert.IsNotNull(savedCompany);
            Assert.AreSame(company.Address, address);
            Assert.AreEqual(savedCompany.Address, address);
            Assert.AreEqual(company.Id, savedCompany.Id);
            Assert.AreEqual(company.Name, savedCompany.Name);
            Assert.AreEqual(company, savedCompany);
            Assert.AreNotSame(company, savedCompany);

            customer.Delete();
            Assert.IsNullOrEmpty(customer.Id);
            Assert.IsNull(Customer.Find(customer.Id));

            company.Delete();
            Assert.IsNullOrEmpty(company.Id);
            Assert.IsNull(Company.Find(company.Id));
        }
    }
```

**Comment by rprouse** [comment]
I tested in Visual Studio 2019. Tests are discovered, I think by Visual Studio, not by the adapter, but tests are not run. In the output window, it looks like only the 3.11 version of the adapter is being called.

**Comment by rprouse** [comment]
I managed to get it to work. To do so, I had to go to **Tools | Options | Test | General** and uncheck the Active Solution performance checkbox. I then had to rebuild the solution. Once I did that, tests appeared and the NUnit 2.1.1.0 adapter started being listed in the output window. I believe the VSIX would also work.

![image](https://user-images.githubusercontent.com/493828/50037667-eb95fd00-ffe1-11e8-954f-878eeca330e9.png)

If we want to fix this issue, I think we will need to move the adapter into the bin directory from the NuGet package like we did with the NUnit 3 adapter.

**Comment by DanJ210** [comment]
Thanks @CharliePoole for the insight, I think they have been discovered by the test explorer because they would show up but not execute.

However, the solution that @rprouse suggested worked. I turned off the setting outlined above and I'm now stepping through the tests. So I'll use this solution for now. I was given specific instructions to change nothing in the test case given so I have to use what's there but I'm sure that's not what they use in their main day.

**Comment by vaindil** [comment]
I don't know if my issue is the same, but unchecking that box did not work for me. All of my tests are discovered by the Test Explorer, but when I try to run any of them I get "No test matches the given testcase filter...". I'm using the VSIX adapter.

```
[2/20/2019 1:00:51 PM Informational] ------ Run test started ------
[2/20/2019 1:00:54 PM Informational] NUnit VS Adapter 2.1.1.0 executing tests is started
[2/20/2019 1:00:59 PM Informational] Loading tests from C:\fakepath\bin\Debug\UnitTests.fakename.dll
[2/20/2019 1:01:01 PM Warning] Dependent Assembly  of C:\fakepath\bin\Debug\UnitTests.fakename.dll not found. Can be ignored if not a NUnit project.
[2/20/2019 1:01:01 PM Informational] NUnit VS Adapter 2.1.1.0 executing tests is finished
[2/20/2019 1:01:01 PM Warning] No test matches the given testcase filter `FullyQualifiedName=UnitTests.fakename.DatabaseTests.TablesExist` in C:\fakepath\bin\Debug\UnitTests.fakename.dll
[2/20/2019 1:01:01 PM Informational] ========== Run test finished: 0 run (0:00:09.9752465) ==========
```

**Comment by OsirisTerje** [comment]
@vaindil   Ok, I'll try to take a look at it.  If you by any chance have a small repro that compiles "as is", it would be awesome.

**Comment by workgroupengineering** [comment]
i have similar issue.

My configuration is

Visual studio 2017 9.11
NUnit VS Adapter 2.1.1.0 
NUnit  2.7.0
Microsoft.NET.Test.Sdk 15.9.0
Test project target .NET 3.5
The adapter is deployed by nuget

When try discovery the test in output window show this messages:
``
[13/05/2019 11:11:48 Informational] NUnit VS Adapter 2.1.1.0 executing tests is started
[13/05/2019 11:11:49 Informational] Loading tests from C:\GitHub\FatturaElettronica.NET\src\Test(3.5)\bin\Debug\net35\FatturaElettronica.Test.dll
[13/05/2019 11:11:49 Error] Exception thrown executing tests in C:\GitHub\FatturaElettronica.NET\src\Test(3.5)\bin\Debug\net35\FatturaElettronica.Test.dll
[13/05/2019 11:11:49 Error] System.InvalidOperationException: Operazione non valida a causa dello stato corrente dell'oggetto.
   in Mono.Cecil.ModuleDefinition.ProcessDebugHeader()
   in Mono.Cecil.ModuleReader.ReadSymbols(ModuleDefinition module, ReaderParameters parameters)
   in Mono.Cecil.ModuleReader.CreateModuleFrom(Image image, ReaderParameters parameters)
   in Mono.Cecil.ModuleDefinition.ReadModule(String fileName, ReaderParameters parameters)
   in NUnit.VisualStudio.TestAdapter.NavigationDataProvider.CacheTypes(String assemblyPath)
   in NUnit.VisualStudio.TestAdapter.NavigationDataProvider.GetNavigationData(String className, String methodName)
   in NUnit.VisualStudio.TestAdapter.TestConverter.MakeTestCaseFromNUnitTest(ITest nunitTest)
   in NUnit.VisualStudio.TestAdapter.TestConverter.ConvertTestCase(ITest test)
   in NUnit.VisualStudio.TestAdapter.AssemblyRunner.AddTestCases(ITest test)
   in NUnit.VisualStudio.TestAdapter.AssemblyRunner.AddTestCases(ITest test)
   in NUnit.VisualStudio.TestAdapter.AssemblyRunner.AddTestCases(ITest test)
   in NUnit.VisualStudio.TestAdapter.AssemblyRunner.AddTestCases(ITest test)
   in NUnit.VisualStudio.TestAdapter.AssemblyRunner.AddTestCases(ITest test)
   in NUnit.VisualStudio.TestAdapter.AssemblyRunner.TryLoadAssembly()
   in NUnit.VisualStudio.TestAdapter.AssemblyRunner.RunAssembly(IFrameworkHandle testLog)
``
I've tried all your suggestions, but I haven't solved them.

I tried to downgrade the adapter to 2.0.0.0.
The tests are discovered.

**Comment by OsirisTerje** [comment]
@workgroupengineering Can you upload a small repro ?

**Comment by workgroupengineering** [comment]
@OsirisTerje [Here](https://github.com/workgroupengineering/Issue178) is an example repository.

**Comment by OsirisTerje** [comment]
@workgroupengineering Thanks for the repo.  The issue is caused by using the new SDK style project format.  If you create the project with the Traditional style project format (either packages.config or package reference, both should work), then it works.  I tested it out with such a project and then it finds the adapter and runs correctly.  
There is something in the way the new VS works with the new project format causing this issue.  I've tried to manually check if I get it right, but not.  The easiest work around right now is to use the Traditional Project format.  
Btw : What is the reason for using NUnit2 and not NUnit3 ?

**Comment by workgroupengineering** [comment]
Thanks @OsirisTerje,
it is strange that it works with version 2.0.0 and not with 2.1.1.
I use NUnit 2.0 because NUnit3 is not compatible with .Net 3.5

**Comment by OsirisTerje** [comment]
NUnit3 is compatible with .net 3.5.  It is not compatible with 2.0, but 3.5 works.    And, I just checked it again on the project, updated to 3.11 framework and 3.13 adapter, and it works.

Also, the test projects could well be running under a higher framework, even if the SUT is targeting 3.5.

**Comment by OsirisTerje** [comment]
@workgroupengineering  When you say it works with 2.0 and not 2.1, that is the framework, which should not affect this at all.  Did you use the same adapter version in both cases ?

**Comment by rprouse** [comment]
> I use NUnit 2.0 because NUnit3 is not compatible with .Net 3.5

@workgroupengineering why do you say that NUnit 3 is not compatible with .NET 3.5? NUnit 3 has always had .NET 3.5 targets. In fact, until the upcoming 3.12 release, it still has .NET 2.0 targets.

**Comment by workgroupengineering** [comment]
Hi @rprouse 
When I tried in multi target project, Visual Studio tells me that the target was invalid.
I will try again.

**Comment by rprouse** [comment]
Thanks @workgroupengineering. It should work, our NUnit tests target everything from .NET 3.5 to .NET Core and run fine. If you can create a repro for us, we can look at it for you.

**Comment by OsirisTerje** [comment]
@rprouse  See my answer above, https://github.com/nunit/nunit-vs-adapter/issues/178#issuecomment-491889818, and if you want to have another look, check his repro here https://github.com/workgroupengineering/Issue178  

I have also forked his repro and added the Traditional style, see https://github.com/OsirisTerje/Issue178, branch FixNUnit.   :-)

**Comment by workgroupengineering** [comment]
@rprouse after update VS at 15.9.2 the issue is resolved

**Comment by OsirisTerje** [comment]
@workgroupengineering @DanJ210  Just released 2.2 of the adapter.  It should work with these now.  See details here:  http://hermit.no/new-version-of-the-nunit-2-adapter-supporting-visual-studio-2019/
